charts/testkube-enterprise/installer_enterprise.sh

```diff
@@ -130,15 +130,55 @@ request_license() {
     parse_license_key
 }
 
+# Function to calculate distinct and persistent machine ID
+get_machine_id() {
+    local machine_id
+
+    # Check if /etc/machine-id exists (common on most modern Linux distributions)
+    if [ -f /etc/machine-id ]; then
+        machine_id=$(cat /etc/machine-id)
+    # Check if dmidecode is available (common on Linux)
+    elif command -v dmidecode >/dev/null 2>&1; then
+        machine_id=$(sudo dmidecode -s system-uuid)
+    # Check if system_profiler command is available (common on macOS)
+    elif command -v system_profiler >/dev/null 2>&1; then
+        machine_id=$(system_profiler SPHardwareDataType | awk '/UUID/ {print $3}')
+    # Check if ioreg command is available (alternative for macOS)
+    elif command -v ioreg >/dev/null 2>&1; then
+        machine_id=$(ioreg -rd1 -c IOPlatformExpertDevice | grep -i "UUID" | awk -F '"' '/UUID/{print $4}')
+    # Check if hostname command is available (fallback)
+    elif command -v hostname >/dev/null 2>&1; then
+        machine_id=$(hostname)
+    else
+        machine_id=""
+    fi
+
+    # If machine ID is not empty, convert it to a UUID
+    if [ -n "$machine_id" ]; then
+        # Remove whitespace and convert to lowercase
+        machine_id=$(echo "$machine_id" | tr -d '[:space:]' | tr '[:upper:]' '[:lower:]')
+        
+        # If the length is not standard UUID length (32 characters), add dashes
+        if [ ${#machine_id} -eq 32 ]; then
+            machine_id=$(echo "$machine_id" | sed 's/\(........\)\(....\)\(....\)\(....\)\(............\)/\1-\2-\3-\4-\5/')
+        fi
+    else
+        machine_id=$(uuidgen 2>>/dev/null || $$ )
+    fi
+
+    echo "$machine_id"
```

Comment:
`echo` will print the machine_id at the stdout for user.

To keep good user experience we have to use the logger and just send that to the log file.

For instance:
`log 0 "DEBUG" $NC "Obtained Machine ID: $machile_id"`

Reply:
In other words, run the script and ensure no output of this function is visible at the stdout by the user running the installer.

If you think any log is required to understand how the machine was obtained, you can use the "log" function or redirect commands output to "$LOG_FILE"

Reply:
@nicufk just told me that this `echo` is the return of the function. I didn't realize about it. So, resolving this conversation. Thanks Nico!